File: d3/p1/p1.php
<?php
$text = file_get_contents('in.txt');
preg_match_all('/mul\(\d{1,3},\d{1,3}\)/', $text, $matches);
$sum = 0;
foreach ($matches[0] as $match) {
    $sum += (int)(substr(explode(",", $match)[0], 4)) * (int)(substr(explode(",", $match)[1], 0, -1));
}
echo $sum . "\n";
?>

File: d3/p1/p1.py
import re
intxt = 'in.txt'
with open(intxt, 'r') as file:
    text = file.read()
matches = re.findall(r"mul\(\d{1,3},\d{1,3}\)", text)
sum = 0
for mul in matches:
    sum += int(mul.split(",")[0][4:]) * int(mul.split(",")[1][:-1])
print(sum)